guides/assets/javascripts/guides.js

```diff
@@ -102,5 +103,91 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var mainColElems = Array.from(document.getElementById("mainCol").children);
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var matchingNavLink = function (elem) {
+      if(!elem) return;
+      var index = mainColElems.indexOf(elem);
+
+      var match;
+      while (index >= 0 && !match) {
+        var link = mainColElems[index].querySelector("a");
+        if (link) {
+          match = document.querySelector('#subCol [href="' + link.getAttribute("href") + '"]');
+        }
+        index--;
+      }
+      return match;
+    }
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
+      removeHighlight();
+      if (!elem) return;
+      elem.classList.add("active");
+      if (window.innerWidth > DESKTOP_THRESHOLD) {
+        elem.scrollIntoView({ block: 'center', inline: 'end' });
+      }
+    }
+
+    var resetNavPosition = function () {
+      var chapters = document.querySelector(".chapters");
+      chapters?.scroll({ top: 0 });
+    }
+
+    var belowBottomHalf = function (i) {
+      return i.boundingClientRect.bottom > (i.rootBounds.bottom + i.rootBounds.top) / 2;
+    }
+
+    var prevElem = function (elem) {
+      var index = mainColElems.indexOf(elem);
+      if (index <= 0) {
+        return null;
+      }
+      return mainColElems[index - 1];
+    }
+
+    var PAGE_LOAD_BUFFER = 1000;
+
+    var navHighlight = function (entries) {
+      entries.forEach(function (entry) {
+        if (entry.time < PAGE_LOAD_BUFFER && !entry.isIntersecting) {
+          return;
+        }
+        else if (belowBottomHalf(entry) && !entry.isIntersecting) {
+          updateHighlight(matchingNavLink(prevElem(entry.target)));
+        } else if (entry.isIntersecting) {
+          updateHighlight(matchingNavLink(entry.target));
+        }
```

Comment:
The 3 if..else here all rely on `entry.isIntersecting`, maybe they could be simplified by an external condition on that, something like this:

```suggestion
        if (entry.isIntersecting) {
          updateHighlight(matchingNavLink(entry.target));
        } else {
          if (entry.time < PAGE_LOAD_BUFFER) {
            return;
          } else if (belowBottomHalf(entry)) {
            updateHighlight(matchingNavLink(prevElem(entry.target)));
          }
       }
```

And looking at that, perhaps we could even group the two last conditionals, since we wouldn't need to return to skip that loop? So it'd be condensed to something like this:

```suggestion
        if (entry.isIntersecting) {
          updateHighlight(matchingNavLink(entry.target));
        } else if (entry.time >= PAGE_LOAD_BUFFER && belowBottomHalf(entry)) {
          updateHighlight(matchingNavLink(prevElem(entry.target)));
       }
```

Totally untested though 😅